test/unit/spec/util/networkmonitor.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const NetworkMonitor = require('../../../../lib/util/networkmonitor');
+
+class MockConnection {
+  constructor() {
+    this.type = 'whiz';
+    this.listeners = {};
+  }
+  addEventListener(event, listener) {
+    this.listeners[event] = listener;
+  }
+  removeEventListener(event) {
+    delete this.listeners[event];
+  }
+  emit(event) {
+    if (this.listeners[event]) {
+      this.listeners[event]();
+    }
+  }
+}
+
+class MockWindow {
+  constructor() {
+    this.listeners = {};
+  }
+  addEventListener(event, listener) {
+    this.listeners[event] = listener;
+  }
+  removeEventListener(event) {
+    delete this.listeners[event];
+  }
+  emit(event) {
+    if (this.listeners[event]) {
+      this.listeners[event]();
+    }
+  }
+}
+
+describe('NetworkMonitor', () => {
+  const onNetworkChangedCalled = sinon.spy();
+
+  describe('constructor', () => {
+    it('should return an instance of NetworkMonitor', () => {
+      const networkMonitor = new NetworkMonitor();
+      assert(networkMonitor instanceof NetworkMonitor);
+    });
+
+    [true, false].forEach(bool => {
+      it(`should return ${bool} when accessing isOnline`, () => {
+        const navigator = { onLine: bool };
+        const networkMonitor = new NetworkMonitor(onNetworkChangedCalled, { navigator });
+        assert.equal(networkMonitor.isOnline, bool);
+      });
+    });
+
+    [{ connection: { type: 'wifi' } }, { connection: {} }, {}].forEach(el => {
```

Comment:
@PikaJoyce Since you have created the MockConnection and MockWindow classes, please use them everywhere possible.